sirepo/pkcli/admin.py

```diff
@@ -49,20 +49,22 @@ def _audit_user(uid, proprietary_sim_types):
                 )
 
     def _link_or_unlink_proprietary_files(sim_type, should_link):
-        b = sim_data.get_class(sim_type).FLASH_RPM_FILENAME
-        p = simulation_db.simulation_lib_dir(sim_type).join(b)
-        if not should_link:
-            pkio.unchecked_remove(p)
-            return
-        try:
-            s = proprietary_code_dir(sim_type).join(b)
-            assert s.check(file=True), f'{s} not found'
-            p.mksymlinkto(
-                s,
-                absolute=False,
-            )
-        except py.error.EEXIST:
-            pass
+        for f in proprietary_code_dir(sim_type).listdir(
+                fil=lambda x: x.check(file=True),
+                sort=True,
+        ):
+            p = simulation_db.simulation_lib_dir(sim_type).join(f.basename)
+            if not should_link:
+                pkio.unchecked_remove(p)
+                return
+            try:
+                assert f.check(file=True), f'{f} not found'
```

Comment:
It could be a directory, no?

Reply:
It can't. [job.\_\_init\_\_.make_lib_dir_symlink](https://github.com/radiasoft/sirepo/blob/00903ca4af2b437c4de426a53cfab1f8b79638eb/sirepo/job_driver/__init__.py#L135) creates a list of the files in the simulation_lib_dir (their basenames). [sim_data.\_\_init\_\_._lib_file_abspath](https://github.com/radiasoft/sirepo/blob/00903ca4af2b437c4de426a53cfab1f8b79638eb/sirepo/sim_data/__init__.py#L568) then pulls those files over (based on their basename). We don't handle any sort of directory structure. We could change the libFileList to have the relative path to files so they could be nested in directories? This would mean the run_dir might not have files in a flat structure. I'd need to look at that code more to see how it is impacted.

Reply:
ok, leave it as is.